hrm-service/service-tests/csam-reports.test.ts

```diff
@@ -0,0 +1,187 @@
+
+const server = app.listen();
+const request = supertest.agent(server);
+
+const { accountSid } = mocks;
+// const workerSid = 'worker-sid';
+/**
+ * Notes: I think these should belong in the mocks.js. I will move them once I finalize this PR. I understand I have to destroy them, and I was hoping you can point me to the best practise to do this.
+ */
+
+const csamReport1 = {
+  csamReportId: 'csam-report-id',
+  twilioWorkerId: 'worker-sid',
+};
+const csamReport2 = {
+  csamReportId: 'csam-report-id',
+  twilioWorkerId: null,
+};
+const csamReport3 = {
+  csamReportId: null,
+  twilioWorkerId: 'worker-sid',
+};
+const csamReport4 = {
+  csamReportId: 'csam-report-id',
+  twilioWorkerId: 'worker-sid',
+  contactId: 1234,
+};
+
+const headers = {
+  'Content-Type': 'application/json',
+  Authorization: `Basic ${Buffer.from(process.env.API_KEY).toString('base64')}`,
+};
+
+const csamReports2DestroyQuery = {
+  where: {
+    accountSid: {
+      [Sequelize.Op.and]: [accountSid],
+    },
+  },
+};
+
+beforeAll(async () => {
+  await CSAMReport.destroy(csamReports2DestroyQuery);
+});
+
+afterAll(done => {
+  server.close(() => {
+    CSAMReport.destroy(csamReports2DestroyQuery).then(() => {
+      console.log('csam reports test cleaned up.');
+      done();
+    });
+  });
+});
+
+describe('/csamReports route', () => {
+  const route = `/v0/accounts/${accountSid}/csamReports`;
+
+  // const csamReportId = 'csam-report-id';
+
+  const expected = {
+    id: expect.anything(),
+    accountSid: accountSid,
+    csamReportId: 'csam-report-id',
+    twilioWorkerId: 'worker-sid',
+    contactId: null,
+    updatedAt: expect.toParseAsDate(),
+    createdAt: expect.toParseAsDate(),
+  };
+
+  describe('POST', () => {
+    test('should return 401', async () => {
+      const response = await request.post(route).send(csamReport1);
+
+      expect(response.status).toBe(401);
+      expect(response.body.error).toBe('Authorization failed');
+    });
+
+    test('should return 200', async () => {
+      const response = await request
+        .post(route)
+        .set(headers)
+        .send(csamReport1);
+
+      expect(response.status).toBe(200);
+      expect(response.body).toStrictEqual(expected);
+
+      const reportFromDB = await CSAMReportController.getCSAMReport(response.body.id, accountSid);
+      /**
+       * Notes: console logging reportFromDB is not returning anything in console. I am thinking the 2 statements below are not being executed.
+       *
+       */
+      expect(reportFromDB).toBeDefined();
+      expect(reportFromDB.csamReportId).toEqual(expected.csamReportId);
+    });
+
+    test('unsure - when request is missing twilioWorkerId', async () => {
+      const response = await request
+        .post(route)
+        .set(headers)
+        .send(csamReport2);
+      expect(response.status).toBe(200);
+      expect(response.body.twilioWorkerId).toEqual('');
+      /**
```

Comment:
If that's how it behaves, just verify that behaviour, this task is to write tests to verify current behaviour - if that behaviour needs changing that's a separate task

We do need to verify that nothing is actually added to the DB though